src/new-client/src/components/DraggableTearOut/DraggableTearOut.tsx

```diff
@@ -0,0 +1,120 @@
+import styled from "styled-components"
+import { RefObject, useState } from "react"
+import { tearOutSection, Section } from "@/App/TearOutSection/state"
+import { canDrag } from "@/components/DraggableTearOut/canDrag"
+import { tearOut } from "@/App/LiveRates/Tile/TearOut/state"
+
+export const DragWrapper = styled.div`
+  height: 100%;
+`
+
+type DraggableSectionTearOutProps = React.PropsWithChildren<{
+  section: Section
+  disabled?: boolean
+}>
+
+type DraggableTileTearOutProps = React.PropsWithChildren<{
+  symbol: string
+  tileRef: RefObject<HTMLDivElement>
+  disabled?: boolean
+}>
+
+type DraggableTearOutGenericProps = React.PropsWithChildren<{
+  dragHandler: () => void
+}>
+
+export const DraggableSectionTearOut: React.FC<DraggableSectionTearOutProps> = (
+  props: DraggableSectionTearOutProps,
+) => {
+  const draggable = canDrag && !props.disabled
+  const dragHandler = () => {
+    tearOutSection(true, props.section)
+  }
+  return (
+    <>
+      {draggable ? (
+        <DraggableTearOut dragHandler={dragHandler}>
+          {props.children}
+        </DraggableTearOut>
+      ) : (
+        props.children
+      )}
+    </>
+  )
+}
+
+export const DraggableTileTearOut: React.FC<DraggableTileTearOutProps> = (
+  props: DraggableTileTearOutProps,
+) => {
+  const draggable = canDrag && !props.disabled
+  const dragHandler = () => {
+    tearOut(props.symbol!, true, props.tileRef?.current!)
```

Comment:
minor - symbol and `tileRef` are mandatory props, so no need for `!` and `?`

Reply:
Fixed